### ISSUE TITLE
*NoOpTracer does not implement vm.EVMLogger (missing CaptureTxEnd method)

### ISSUE THREAD

**Issue by Percronos** [tharsis/ethermint#1130]
Hey team,

When go build a small script that calls (indirectly) github.com/tharsis/ethermint v0.15.0, it fails:

# github.com/tharsis/ethermint/x/evm/types
../../../go/pkg/mod/github.com/tharsis/ethermint@v0.15.0/x/evm/types/tracer.go:43:10: cannot use NewNoOpTracer() (value of type *NoOpTracer) as type vm.EVMLogger in return statement:
        *NoOpTracer does not implement vm.EVMLogger (missing CaptureTxEnd method)
../../../go/pkg/mod/github.com/tharsis/ethermint@v0.15.0/x/evm/types/tracer.go:128:22: cannot use &NoOpTracer{} (value of type *NoOpTracer) as type vm.EVMLogger in variable declaration:
        *NoOpTracer does not implement vm.EVMLogger (missing CaptureTxEnd method)

The exact same script on an other VM is working fine.
Thanks for your help.

Best,

**Comment by fedekunze** [comment]
I don't think this is a bug, we're using Geth `v1.10.16` and it implements the interface, which go-ethereum version are you using?

**Comment by fedekunze** [comment]
Might be related to upgrading to the latest geth version (#1132)

**Comment by Percronos** [comment]
Hey,

Thanks for your answer.
I'm using v1.10.16 (through Cronos).

I've modified the return on tracer.go:43:10 and commented tracer.go:128:22, as I'm not using them and it's working (not a fix, but the problem is currently avoided).
